crates/oxc_linter/src/rules/react/checked_requires_onchange_or_readonly.rs

```diff
@@ -0,0 +1,270 @@
+                if !is_create_element_call(call_expr) {
+                    return;
+                }
+
+                let Some(Argument::Expression(Expression::StringLiteral(element_name))) =
+                    call_expr.arguments.first()
+                else {
+                    return;
+                };
+
+                if element_name.value != "input" {
+                    return;
+                }
+
+                let Some(Argument::Expression(Expression::ObjectExpression(obj_expr))) =
+                    call_expr.arguments.get(1)
+                else {
+                    return;
+                };
+
+                let prop_names: Vec<String> = obj_expr
+                    .properties
+                    .iter()
+                    .filter_map(|attr| {
+                        if let ObjectPropertyKind::ObjectProperty(obj_prop) = attr {
+                            let name = &obj_prop.key.name()?;
+                            if is_target_prop(name) {
+                                return Some(name.to_string());
+                            }
+                        }
+                        None
+                    })
+                    .collect();
+
+                if let Some(span) = obj_expr.properties.iter().find_map(|prop| {
+                    if let ObjectPropertyKind::ObjectProperty(prop) = prop {
+                        if prop.key.is_specific_static_name("checked") {
+                            return Some(prop.span);
+                        }
+                    }
+
+                    None
+                }) {
+                    self.check_attributes_and_report(ctx, &prop_names, span);
+                }
+            }
+            _ => {}
+        }
+    }
+
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let value = value.as_array().and_then(|arr| arr.first()).and_then(|val| val.as_object());
+
+        Self {
+            ignore_missing_properties: value
+                .and_then(|val| {
+                    val.get("ignoreMissingProperties").and_then(serde_json::Value::as_bool)
+                })
+                .unwrap_or(false),
+            ignore_exclusive_checked_attribute: value
+                .and_then(|val| {
+                    val.get("ignoreExclusiveCheckedAttribute").and_then(serde_json::Value::as_bool)
+                })
+                .unwrap_or(false),
+        }
+    }
+}
+
+impl CheckedRequiresOnchangeOrReadonly {
+    fn check_attributes_and_report(&self, ctx: &LintContext, props: &[String], span: Span) {
+        if !self.ignore_exclusive_checked_attribute && props.contains(&"defaultChecked".to_string())
+        {
+            ctx.diagnostic(CheckedRequiresOnchangeOrReadonlyDiagnostic::ExclusiveCheckedAttribute(
+                span,
+            ));
+        }
+
+        if !(self.ignore_missing_properties
+            || props.contains(&"onChange".to_string())
+            || props.contains(&"readOnly".to_string()))
```

Comment:
```suggestion
            || props.contains("onChange")
            || props.contains("readOnly"))
```

I think we don't need to use `to_string`

Reply:
Thank you.
When `to_string` was removed, the following error was displayed 🤔

> mismatched types expected reference `&std::string::String` found reference `&'static str`

Reply:
Try to see if you can change the type of prop_names to Vec<&'a str>.

Reply:
After attempting the fix, the error within `check_attributes_and_report` disappeared.
Instead, the following error started occurring in the caller of `check_attributes_and_report`:

> mismatched types expected struct std::vec::Vec<&str> found reference &std::vec::Vec<std::string::String>

Since the [prop_names in the caller](https://github.com/oxc-project/oxc/pull/2754/files#diff-948df61eb03dd7247d0b26e3583f7e3d4ded6b1606bf637b9f02821beabafef8R130-R142) is of type Vec\<String>, I attempted to modify it to Vec<&str>, which resulted in the following error:

> cannot return value referencing temporary value returns a value referencing data owned by the current function

```diff rust
- let prop_names: Vec<String> = obj_expr
+ let prop_names: Vec<&str> = obj_expr
    .properties
    .iter()
    .filter_map(|attr| {
        if let ObjectPropertyKind::ObjectProperty(obj_prop) = attr {
            let name = &obj_prop.key.name()?;
            if is_target_prop(name) {
-                 return Some(name.to_string());
+                 return Some(name.as_str());
            }
        }
        None
    })
    .collect();
```

Reply:
Oh, yes, we can't do that. Thank you for your try!